homeassistant/components/awair/sensor.py

```diff
@@ -147,6 +151,7 @@ def available(self) -> bool:
     def native_value(self) -> float:
         """Return the state, rounding off to reasonable values."""
         state: float
+        assert self._air_data
```

Comment:
Not sure if this is OK, or if `self._air_data` can be `None` and that should be handled.

Reply:
Codeowner should answer this, or we should add a guard.

Reply:
Well, at L238 we see:

```python
    def _air_data(self) -> AirData | None:
        """Return the latest data for our device, or None."""
        result: AwairResult | None = self.coordinator.data.get(self._device.uuid)
        if result:
```

I don't think it's wise to assert that this is always present, especially since there are guards present elsewhere.

Reply:
As @MartinHjelmare explained the assert is there to tell MyPy we think `self._air_data` is always valid (that's basically what the current code assumes since it doesn't check for `None`).

What do you suggest to do if `self._air_data` is in fact None? Should the entity be set to unavailable, unknown state or something else?

Reply:
I would say it should be set to `unavailable`, that's a good point.